Open Judge System/Services/OJS.Services/Business/ParticipantScores/ParticipantScoresBusinessService.cs

```diff
@@ -377,37 +372,60 @@ private double CalculateTimeInMinutesFromParticipationStartToLastSubmission(Part
                 .ToDictionary(k => k.Key.ToString(), v => v.Value);
         }
 
-        private void NormalizeSubmissionPoints() =>
-            this.submissionsData
+        private void NormalizeSubmissionPoints()
+        {
+            var itemsToTake = 300;
+            var itemsToSkip = 0;
+            var submissionsWithProblemMaxPoints = this.submissionsData
                 .GetAllHavingPointsExceedingLimit()
                 .Select(s => new
                 {
                     Submission = s,
                     ProblemMaxPoints = s.Problem.MaximumPoints,
                 })
-                .ToList()
-                .ForEach(x =>
+                .ToList();
+            while (itemsToSkip < submissionsWithProblemMaxPoints.Count)
+            {
+                using (var scope = TransactionsHelper.CreateLongRunningTransactionScope())
                 {
-                    x.Submission.Points = x.ProblemMaxPoints;
+                    submissionsWithProblemMaxPoints.Skip(itemsToSkip).Take(itemsToTake).ToList().ForEach(x =>
+                    {
+                        x.Submission.Points = x.ProblemMaxPoints;
 
-                    this.submissionsData.Update(x.Submission);
-                });
+                        this.submissionsData.Update(x.Submission, true);
+                    });
+                    
+                    this.submissionsData.SaveChanges();
+                    scope.Complete();
```

Comment:
There is no point of executing this in a transaction scope as `this.submissionsData.SaveChanges();` will be a transaction for all the updates before it.

Reply:
fixed